revops/revops.js

```diff
@@ -51,16 +51,26 @@ async function populateBlogPosts() {
   const request = new Request("blog-posts.json");
 
   const response = await fetch(request);
-  blogPosts = await response.json();
+  var unorderedBlogPosts = await response.json();
+
+  blogPosts = unorderedBlogPosts.sort(function (a, b) {
+    if (new Date(a.Date) == new Date(b.Date)) return 0;
+    if (new Date(a.Date) < new Date(b.Date)) return -1;
+    if (new Date(a.Date) > new Date(b.Date)) return 1;
```

Comment:
Code style:

I like the if-return style here. In order to keep the code readable (and actually more performant!) I would suggest reducing the complexity by extracting variables.

I would also encourage being consistent with indentation levels and indications of nested blocks. That is, *always* use curly braces when specifying a block, and *always* indent the same level. This makes it easier to quickly parse code, as the indentation levels give a quick indication of what is related and how.

```js
const aDate = new Date(a.Date);
const bDate = new Date(b.Date);
if (aDate == bDate) {
  return 0;
}
if(aDate < bDate) {
  return -1;
}
if(aDate > bDate) {
  return 1;
}
```

This is *more lines* but that's not a problem. Lines are free.

On the other hand it is more *honest* about the complexity of the function, instead of trying to hide the complexity by fitting as much as possible into each line.

It's like the difference between writing long, convoluted paragraph-length sentences to seem clever, or breaking writing into shorter, more easily digestible sentences. When the goal is only clarity of communication, it is *always* better to err on the side of shorter and clearer sentences.

Reply:
Yes, makes total sense, fixed